backend/src/main/java/gov/cdc/usds/simplereport/service/PatientBulkUploadService.java

```diff
@@ -35,186 +33,128 @@
 
-    Instant startTime = Instant.now();
+    Optional<Facility> facility =
+        Optional.ofNullable(facilityId).map(_organizationService::getFacilityInCurrentOrg);
 
-    int rowNumber = 0;
     while (valueIterator.hasNext()) {
-      final Instant rowStartTime = Instant.now();
-      final Map<String, String> row = getNextRow(valueIterator);
-      rowNumber++;
+      final Map<String, String> row = CsvValidatorUtils.getNextRow(valueIterator);
+
       try {
-        Optional<Facility> facility =
-            Optional.ofNullable(facilityId).map(organizationService::getFacilityInCurrentOrg);
 
-        String zipCode = getRow(row, "ZipCode", true);
-        if (!zipCode.matches(ZIP_CODE_REGEX)) {
-          throw new IllegalArgumentException("Invalid zip code");
-        }
+        PatientUploadRow extractedData = new PatientUploadRow(row);
 
+        // Fetch address information
         StreetAddress address =
-            addressValidationService.getValidatedAddress(
-                getRow(row, "Street", true),
-                getRow(row, "Street2", false),
-                getRow(row, "City", false),
-                getRow(row, "State", true),
-                zipCode,
+            _addressValidationService.getValidatedAddress(
+                extractedData.getStreet().getValue(),
+                extractedData.getStreet2().getValue(),
+                extractedData.getCity().getValue(),
+                extractedData.getState().getValue(),
+                extractedData.getZipCode().getValue(),
                 null);
 
-        var firstName = parseString(getRow(row, "FirstName", true));
-        var lastName = parseString(getRow(row, "LastName", true));
-        var dob = parseUserShortDate(getRow(row, "DOB", true));
-        var country = parseString(getRow(row, "Country", false));
+        String country = "USA";
 
-        if (country == null) {
-          country = "USA";
-        }
-
-        if (personService.isDuplicatePatient(firstName, lastName, dob, org, facility)) {
+        if (_personService.isDuplicatePatient(
+            extractedData.getFirstName().getValue(),
+            extractedData.getLastName().getValue(),
+            parseUserShortDate(extractedData.getDateOfBirth().getValue()),
+            org,
+            facility)) {
           continue;
         }
 
-        personService.addPatient(
+        _personService.addPatient(
             facilityId,
             null, // lookupID
-            firstName,
-            parseString(getRow(row, "MiddleName", false)),
-            lastName,
-            parseString(getRow(row, "Suffix", false)),
-            dob,
+            extractedData.getFirstName().getValue(),
+            extractedData.getMiddleName().getValue(),
+            extractedData.getLastName().getValue(),
+            extractedData.getSuffix().getValue(),
+            parseUserShortDate(extractedData.getDateOfBirth().getValue()),
             address,
             country,
```

Comment:
it looks like we're never allowing the option for another country besides "USA", unless I'm missing a call to `extractedData.getCountry()` or something similar somewhere. Is that correct?

Reply:
That is correct. The reason I did this USA-only version is that as far as I know, we don't have a mapping of country codes on the backend, [only the frontend](https://github.com/CDCgov/prime-simplereport/blob/a476ef8221e6077e026b8628e309fcc245960770/frontend/src/config/constants.ts#L133-L133). Since we don't have anything to validate the country codes against right now, I just defaulted everything to the US.

...but, I just checked to see how many international addresses we actually have in the database, and forgot that Guam has its own country code. So we've got over 30k international patients in SimpleReport already, and I could see Guam wanting to do bulk uploads. I'll add a new commit to include the country validation.
![Screen Shot 2022-11-07 at 9 33 21 AM](https://user-images.githubusercontent.com/80282552/200377142-9570a7f3-c8ef-4446-b9fb-2d12c52c31a3.png)

Reply:
@georgehager this is fixed now!